.github/workflows/links-check.yml

```
@@ -0,0 +1,89 @@
+name: Internal Links Check
+
+on:
+  pull_request:
+    paths:
+      - 'docs/**'
+      - '*.md'
+
+permissions:
+  pull-requests: write
+
+jobs:
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
```

I'd make this change for the `actions/*` actions:

```suggestion
        uses: actions/checkout@main
```